clients/adapter-kafka/src/main/java/org/eclipse/hono/adapter/client/command/kafka/KafkaBasedInternalCommandConsumer.java

```diff
@@ -0,0 +1,237 @@
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.Vertx;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.kafka.admin.KafkaAdminClient;
+import io.vertx.kafka.admin.NewTopic;
+import io.vertx.kafka.client.common.TopicPartition;
+import io.vertx.kafka.client.consumer.KafkaConsumer;
+import io.vertx.kafka.client.consumer.KafkaConsumerRecord;
+
+/**
+ * A Kafka based consumer to receive commands forwarded by the Command Router on the internal command topic.
+ *
+ */
+public class KafkaBasedInternalCommandConsumer implements Lifecycle {
+
+    private static final Logger LOG = LoggerFactory.getLogger(KafkaBasedInternalCommandConsumer.class);
+
+    private static final int NUM_PARTITIONS = 1;
+    private static final short REPLICATION_FACTOR = 1; // TODO use value from adapter configuration instead.
+
+    private final String adapterInstanceId;
+    private final CommandHandlers commandHandlers;
+    private final Tracer tracer;
+    private final KafkaAdminClient adminClient;
+    private final KafkaConsumer<String, Buffer> consumer;
+
+    /**
+     * Creates a consumer.
+     *
+     * @param vertx The Vert.x instance to use.
+     * @param consumerConfigProperties The Kafka consumer config properties.
+     * @param adminClientConfigProperties The Kafka admin client config properties.
+     * @param adapterInstanceId The adapter instance id.
+     * @param commandHandlers The command handlers to choose from for handling a received command.
+     * @param tracer The OpenTracing tracer.
+     * @throws NullPointerException if any of the parameters is {@code null}.
+     */
+    public KafkaBasedInternalCommandConsumer(
+            final Vertx vertx,
+            final KafkaAdminClientConfigProperties adminClientConfigProperties,
+            final KafkaConsumerConfigProperties consumerConfigProperties,
+            final String adapterInstanceId,
+            final CommandHandlers commandHandlers,
+            final Tracer tracer) {
+        Objects.requireNonNull(vertx);
+        Objects.requireNonNull(adminClientConfigProperties);
+        Objects.requireNonNull(consumerConfigProperties);
+        this.adapterInstanceId = Objects.requireNonNull(adapterInstanceId);
+        this.commandHandlers = Objects.requireNonNull(commandHandlers);
+        this.tracer = Objects.requireNonNull(tracer);
+
+        final Map<String, String> adminClientConfig = adminClientConfigProperties.getAdminClientConfig("internal-cmd");
+        adminClient = KafkaAdminClient.create(vertx, adminClientConfig);
+
+        final Map<String, String> consumerConfig = consumerConfigProperties.getConsumerConfig("internal-cmd");
+        // disable auto commit; offsets aren't committed here because topic and consumer are tied to
+        // the particular protocol adapter instance, with topic and consumer becoming obsolete once the
+        // adapter instance is stopped
+        consumerConfig.put("enable.auto.commit", "false");
```

Comment:
How about `ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG` instead of `"enable.auto.commit"`?